lib/npg_pipeline/function/haplotype_caller.pm

```diff
@@ -94,8 +96,23 @@ sub create {
         $chuncking_base_name,
         $product->chunk;
       my $output_path = $product->file_path($out_dir_path, ext => 'g.vcf.gz');
-      my $command = sprintf q{%s %s %s -I %s -O %s -L %s},
-        $self->gatk_cmd, $GATK_TOOL_NAME, $gatk_args, $input_path, $output_path, $region;
+      my $command;
+
+      if ($self->bqsr_enable($product) && $self->bqsr_apply_enable($product)) {
+        # TODO: write proper temporary file handling support
+        my $temp_path = $product->file_path($out_dir_path, ext => 'temp.cram');
+        my $gatk_cmd = sprintf q{%s %s %s -I %s -O %s -L %s},
+          $self->gatk_cmd, $GATK_TOOL_NAME, $gatk_args, $input_path, $temp_path, $region;
+
+        my $bqsr_args = "-R $ref_path --preserve-qscores-less-than 6 --static-quantized-quals 10 --static-quantized-quals 20 --static-quantized-quals 30";
+        my $bqsr_cmd = sprintf q(%s %s %s -I %s -O %s -L %s),
+          $self->gatk_cmd, $GATK_BQSR_TOOL_NAME, $bqsr_args, $temp_path, $output_path, $region;
+        my $rm_cmd = "rm \"$temp_path\" \"$temp_path.crai\"";
+        $command = join ' && ', [$bqsr_cmd, $gatk_cmd, $rm_cmd];
```

Comment:
Do we want to have non-zero exit if the rm command fails?

Reply:
Fixed